README.md
# Yiimp_install_scrypt v0.3 (update October, 2020)

Site : https://easy-client-co.ltd

Discord : https://discord.gg/zcCXjkQ

TUTO Youtube (16.04 / 18.04 - Without SSL) : https://www.youtube.com/watch?v=qE0rhfJ1g2k

Dirty Harry Yiimp (used in this script for Yiimp Installation): https://github.com/NCEX/yiimp

Original Yiimp Installer : https://github.com/cryptopool-builders/multipool_original_yiimp_installer


***********************************

## Install script for yiimp on Ubuntu Server 16.04 / 18.04 (use Dirty Harry's Yiimp)

USE THIS SCRIPT ON FRESH INSTALL UBUNTU Server 16.04 / 18.04 !

Connect on your VPS =>
- apt update
- apt upgrade
- reboot
- adduser pool (pool it's just an example...)
- adduser pool sudo
- su - pool
- sudo apt -y install git
- git clone https://github.com/NCEX/yiimp_install_script.git
- cd yiimp_install_script/
- bash install.sh (DO NOT RUN THE SCRIPT AS ROOT or SUDO)
- At the end, the script REBOOT AUTO to finalize installation...

Finish !
- Go http://xxx.xxx.xxx.xxx or https://xxx.xxx.xxx.xxx (if you have chosen LetsEncrypt SSL). Enjoy !
- Go http://xxx.xxx.xxx.xxx/PoolPortal or https://xxx.xxx.xxx.xxx/PoolPortal to access the Admin Panel.

If you are issue after installation (nginx,mariadb... not found), use this script : (watch the log during installation)


###### :booom: **YOU MUST UPDATE THE FOLLOWING FILES :**
- **/var/web/serverconfig.php :** update this file to include your public ip (line = YAAMP_ADMIN_IP) to access the admin panel (Put your PERSONNAL IP, NOT IP of your VPS). update with public keys from exchanges. update with other information specific to your server..
- **/etc/yiimp/keys.php :** update with secrect keys from the exchanges (not mandatory)
- **If you want change 'PoolPortal' to access the Admin Panel :** Edit this file "/var/web/yaamp/modules/site/SiteController.php" and Line 11 => change 'PoolPortal'


###### :booom: **IMPORTANT** : 

- The configuration of yiimp and coin require a minimum of knowledge in linux
- Your mysql information (login/Password) is saved in **~/.my.cnf**

***********************************

###### This script has an interactive beginning and will ask for the following information :

- Server Name (no http:// or www !!!!! Example : crypto.com OR pool.crypto.com OR 80.41.52.63)
- Are you using a subdomain (mypoolx11.crypto.com)
- Enter support email
- Set stratum to AutoExchange
- Your Public IP for admin access (Put your PERSONNAL IP, NOT IP of your VPS)
- Install Fail2ban
- Install UFW and configure ports
- Install LetsEncrypt SSL

***********************************

**This install script will get you 95% ready to go with yiimp. There are a few things you need to do after the main install is finished.**

While I did add some server security to the script, it is every server owners responsibility to fully secure their own servers. After the installation you will still need to customize your serverconfig.php file to your liking, add your API keys, and build/add your coins to the control panel. 

There will be several wallets already in yiimp. These have nothing to do with the installation script and are from the database import from the yiimp github. 

If you need further assistance we have a small but growing discord channel at https://discord.gg/RCGP4Jh

If this helped you or you feel giving please donate : 
- BTC Donation : 3Ee5UwmuekHD4UkMowMnCKhe9wgUoQpbTf
- ETH Donation : 0x103170406c163a109d81c0cec7374aacb5817cd8
- LTC Donation : MM6y8c3i58kEFhC2JTveF36CYDcGaAdUZk


changelog
Yiimp Install Scrypt v0.3 // 26.10.2020
Thanks to cryptopool-builders !!!!!!!!

Information =>
* Works with Ubuntu Server 16.04 / 18.04
* Update PHP 7.3
* Change Aptitude by APT
* Remove question Panel Admin
* Remove question Time Zone
* Add screen-scrypt to crontab
* Add script to remove all coin (utils/delcoin.sh)

Fix =>
* Fix command cli « Yiimp » : OK
* Fix Backup SQL FrontEnd : OK (/var/yiimp/sauv)
* Fix Permission on some folder

Change =>
* /etc/screen-scrypt.sh = add to crontab (main,loop2,blocks,debug)
* /var/yiimp/sauv =  Backup SQL FrontEnd


conf/getip.sh
#####################################################
# Source https://mailinabox.email/ https://github.com/mail-in-a-box/mailinabox
# Updated by cryptopool.builders for crypto use...
# Modified by Dirty Harry
#####################################################

source /etc/functions.sh
echo
echo
echo -e "$CYAN => Setting our global variables : $COL_RESET"
echo

# If the machine is behind a NAT, inside a VM, etc., it may not know
# its IP address on the public network / the Internet. Ask the Internet
# and possibly confirm with user.
if [ -z "${PUBLIC_IP:-}" ]; then
# Ask the Internet.
GUESSED_IP=$(get_publicip_from_web_service 4)

# On the first run, if we got an answer from the Internet then don't
# ask the user.
if [[ -z "${DEFAULT_PUBLIC_IP:-}" && ! -z "$GUESSED_IP" ]]; then
PUBLIC_IP=$GUESSED_IP

# On later runs, if the previous value matches the guessed value then
# don't ask the user either.
elif [ "${DEFAULT_PUBLIC_IP:-}" == "$GUESSED_IP" ]; then
PUBLIC_IP=$GUESSED_IP
fi

if [ -z "${PUBLIC_IP:-}" ]; then
input_box "Public IP Address" \
"Enter the public IP address of this machine, as given to you by your ISP.
\n\nPublic IP address:" \
"$DEFAULT_PUBLIC_IP" \
PUBLIC_IP

if [ -z "$PUBLIC_IP" ]; then
# user hit ESC/cancel
exit
fi
fi
fi

# Same for IPv6. But it's optional. Also, if it looks like the system
# doesn't have an IPv6, don't ask for one.
if [ -z "${PUBLIC_IPV6:-}" ]; then
	# Ask the Internet.
	GUESSED_IP=$(get_publicip_from_web_service 6)
	MATCHED=0
	if [[ -z "${DEFAULT_PUBLIC_IPV6:-}" && ! -z "$GUESSED_IP" ]]; then
		PUBLIC_IPV6=$GUESSED_IP
	elif [[ "${DEFAULT_PUBLIC_IPV6:-}" == "$GUESSED_IP" ]]; then
		# No IPv6 entered and machine seems to have none, or what
		# the user entered matches what the Internet tells us.
		PUBLIC_IPV6=$GUESSED_IP
		MATCHED=1
	elif [[ -z "${DEFAULT_PUBLIC_IPV6:-}" ]]; then
		DEFAULT_PUBLIC_IP=$(get_default_privateip 6)
	fi

	if [[ -z "${PUBLIC_IPV6:-}" && $MATCHED == 0 ]]; then
		input_box "IPv6 Address (Optional)" \
			"Enter the public IPv6 address of this machine, as given to you by your ISP.
			\n\nLeave blank if the machine does not have an IPv6 address.
			\n\nPublic IPv6 address:" \
			${DEFAULT_PUBLIC_IPV6:-} \
			PUBLIC_IPV6

		if [ ! $PUBLIC_IPV6_EXITCODE ]; then
			# user hit ESC/cancel
			exit
		fi
	fi
fi

# Get the IP addresses of the local network interface(s) that are connected
# to the Internet. We need these when we want to have services bind only to
# the public network interfaces (not loopback, not tunnel interfaces).
# if [ -z "$PRIVATE_IP" ]; then
# DEFAULT_PRIVATE_IP=$(get_default_privateip 4)
# input_box "Private IP Address (Optional)" \
# "Enter the private IP address of this machine, as given to you by your ISP.
# \n\nLeave as your public IP if the machine does not have a private IP address.
# \n\nPrivate IP address:" \
# $DEFAULT_PRIVATE_IP \
# PRIVATE_IP
#
#  if [ -z "$PRIVATE_IP" ]; then
# user hit ESC/cancel
# exit
#  fi
# fi

# Automatic configuration, e.g. as used in our Vagrant configuration.
if [ "$PUBLIC_IP" = "auto" ]; then
# Use a public API to get our public IP address, or fall back to local network configuration.
PUBLIC_IP=$(get_publicip_from_web_service 4 || get_default_privateip 4)
fi
if [ "$PUBLIC_IPV6" = "auto" ]; then
# Use a public API to get our public IPv6 address, or fall back to local network configuration.
PUBLIC_IPV6=$(get_publicip_from_web_service 6 || get_default_privateip 6)
fi

echo -e "$GREEN Done...$COL_RESET"

utils/compile.sh
#!/bin/bash
################################################################################
# Original Author: crombiecrunch
# Current Author: Dirty Harry
# Web: https://easy-client-co.ltd
#
# Program:
#   After entering coin name and github link automatically build coin
#
#
################################################################################
output() {
    printf "\E[0;33;40m"
    echo $1
    printf "\E[0m"
}
displayErr() {
    echo
    echo $1;
    echo
    exit 1;
}
cd ~

if [[ ! -e 'CoinBuilds' ]]; then
        sudo mkdir CoinBuilds
elif [[ ! -d 'CoinBuilds' ]]; then
    output "Coinbuilds already exists.... Skipping" 1>&2
fi

clear
cd CoinBuilds
output "This script assumes you already have the dependicies installed on your system!"
output ""
    read -e -p "Enter the name of the coin : " coin
    read -e -p "Paste the github link for the coin : " git_hub

if [[ ! -e '$coin' ]]; then
    sudo  git clone $git_hub  $coin
elif [[ ! -d ~$CoinBuilds/$coin ]]; then
    output "Coinbuilds/$coin already exists.... Skipping" 1>&2
    output "Can not continue"
    exit 0
fi

cd "${coin}"

if [ -f autogen.sh ]; then
		output " "
        output "Auto Compilation Starting "
		output " "
		output "Starting ./autogen.sh"
		output " "
		sudo chmod +x ./autogen.sh
		sudo ./autogen.sh
		output " "
		output "Starting ./configure"
		output " "
		sudo chmod +x ./configure
		sudo ./configure CPPFLAGS="-I/usr/local/include"
		sudo chmod +x share/genbuild.sh
		output " "
		output "Starting make"
		output " "
        sudo make
		output " "
        output "$coin_name finished and can be found in CoinBuilds/$coin/src/ Make sure you sudo strip Coind and coin-cli if it exists, copy to /usr/bin"
		output " "
else
        cd src
	
if [ -f rpcrawtransaction.cpp ]; then

	sudo sed -i 's/<const\ CScriptID\&/<CScriptID/' rpcrawtransaction.cpp
	sudo sed -i 's/<CScriptID&>/<CScriptID>/g' rpcrawtransaction.cpp
	
if [[ ! -e 'obj' ]]; then
	output " "
	output "Creation directory obj"
	output " "
	sudo mkdir obj
elif [[ ! -d 'obj' ]]; then
    output "Hey the developer did his job" 1>&2
fi

if [ -d 'leveldb' ]; then
	output " "
	output "Compilation libleveldb.a libmemenv.a"
	output " "
	cd leveldb
	sudo chmod +x build_detect_platform
	sudo make clean
	sudo make libleveldb.a libmemenv.a
	cd ..
fi
output " "
output "SRC Compilation Starting "
output " "
sudo make -f makefile.unix
output " "
output "$coin finished and can be found in CoinBuilds/$coin/src/ Make sure you sudo strip Coind and coin-cli if it exists, copy to /usr/bin"
output " "
fi


conf/functions.sh
#!/bin/bash
#####################################################
# Source https://mailinabox.email/ https://github.com/mail-in-a-box/mailinabox
# Updated by cryptopool.builders for crypto use...
# Modified by Dirty Harry
#####################################################

ESC_SEQ="\x1b["
COL_RESET=$ESC_SEQ"39;49;00m"
RED=$ESC_SEQ"31;01m"
GREEN=$ESC_SEQ"32;01m"
YELLOW=$ESC_SEQ"33;01m"
BLUE=$ESC_SEQ"34;01m"
MAGENTA=$ESC_SEQ"35;01m"
CYAN=$ESC_SEQ"36;01m"


function spinner {
 		local pid=$!
 		local delay=0.75
 		local spinstr='|/-\'
 		while [ "$(ps a | awk '{print $1}' | grep $pid)" ]; do
 				local temp=${spinstr#?}
 				printf " [%c]  " "$spinstr"
 				local spinstr=$temp${spinstr%"$temp"}
 				sleep $delay
 				printf "\b\b\b\b\b\b"
 		done
 		printf "    \b\b\b\b"
 }


function hide_output {
		OUTPUT=$(tempfile)
		$@ &> $OUTPUT & spinner
		E=$?
		if [ $E != 0 ]; then
		echo
		echo FAILED: $@
		echo -----------------------------------------
		cat $OUTPUT
		echo -----------------------------------------
		exit $E
		fi

		rm -f $OUTPUT
}


function apt_get_quiet {
		DEBIAN_FRONTEND=noninteractive hide_output sudo apt -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confnew" "$@"
}


function apt_install {
		PACKAGES=$@
		apt_get_quiet install $PACKAGES
}


function ufw_allow {
		if [ -z "$DISABLE_FIREWALL" ]; then
		sudo ufw allow $1 > /dev/null;
		fi
}

function restart_service {
		hide_output sudo service $1 restart
}

## Dialog Functions ##
function message_box {
		dialog --title "$1" --msgbox "$2" 0 0
}

function input_box {
		# input_box "title" "prompt" "defaultvalue" VARIABLE
		# The user's input will be stored in the variable VARIABLE.
		# The exit code from dialog will be stored in VARIABLE_EXITCODE.
		declare -n result=$4
		declare -n result_code=$4_EXITCODE
		result=$(dialog --stdout --title "$1" --inputbox "$2" 0 0 "$3")
		result_code=$?
}

function input_menu {
		# input_menu "title" "prompt" "tag item tag item" VARIABLE
		# The user's input will be stored in the variable VARIABLE.
		# The exit code from dialog will be stored in VARIABLE_EXITCODE.
		declare -n result=$4
		declare -n result_code=$4_EXITCODE
		local IFS=^$'\n'
		result=$(dialog --stdout --title "$1" --menu "$2" 0 0 0 $3)
		result_code=$?
}

function get_publicip_from_web_service {
		# This seems to be the most reliable way to determine the
		# machine's public IP address: asking a very nice web API
		# for how they see us. Thanks go out to icanhazip.com.
		# See: https://major.io/icanhazip-com-faq/
		#
		# Pass '4' or '6' as an argument to this function to specify
		# what type of address to get (IPv4, IPv6).
		curl -$1 --fail --silent --max-time 15 icanhazip.com 2>/dev/null
}

function get_default_privateip {
		# Return the IP address of the network interface connected
		# to the Internet.
		#
		# Pass '4' or '6' as an argument to this function to specify
		# what type of address to get (IPv4, IPv6).
		#
		# We used to use `hostname -I` and then filter for either
		# IPv4 or IPv6 addresses. However if there are multiple
		# network interfaces on the machine, not all may be for
		# reaching the Internet.
		#
		# Instead use `ip route get` which asks the kernel to use
		# the system's routes to select which interface would be
		# used to reach a public address. We'll use 8.8.8.8 as
		# the destination. It happens to be Google Public DNS, but
		# no connection is made. We're just seeing how the box
		# would connect to it. There many be multiple IP addresses
		# assigned to an interface. `ip route get` reports the
		# preferred. That's good enough for us. See issue #121.
		#
		# With IPv6, the best route may be via an interface that
		# only has a link-local address (fe80::*). These addresses
		# are only unique to an interface and so need an explicit
		# interface specification in order to use them with bind().
		# In these cases, we append "%interface" to the address.
		# See the Notes section in the man page for getaddrinfo and
		# https://discourse.mailinabox.email/t/update-broke-mailinabox/34/9.
		#
		# Also see ae67409603c49b7fa73c227449264ddd10aae6a9 and
		# issue #3 for why/how we originally added IPv6.

		target=8.8.8.8

		# For the IPv6 route, use the corresponding IPv6 address
		# of Google Public DNS. Again, it doesn't matter so long
		# as it's an address on the public Internet.
		if [ "$1" == "6" ]; then target=2001:4860:4860::8888; fi

		# Get the route information.
		route=$(ip -$1 -o route get $target | grep -v unreachable)

		# Parse the address out of the route information.
		address=$(echo $route | sed "s/.* src \([^ ]*\).*/\1/")

		if [[ "$1" == "6" && $address == fe80:* ]]; then
		# For IPv6 link-local addresses, parse the interface out
		# of the route information and append it with a '%'.
		interface=$(echo $route | sed "s/.* dev \([^ ]*\).*/\1/")
		address=$address%$interface
		fi

		echo $address

}

utils/packagecompile.sh
    
#!/bin/bash
################################################################################
# Author: Dirty Harry
# Web: https://easy-client-co.ltd
#
# Program:
#   Install needed Package to compile crypto currency
# 
# 
################################################################################
    
    # Installing Package to compile crypto currency
    output " "
    output "Installing needed Package to compile crypto currency"
    output " "
    sleep 3
    
    sudo apt -y install software-properties-common build-essential
    sudo apt -y install libtool autotools-dev automake pkg-config libssl-dev libevent-dev bsdmainutils git cmake libboost-all-dev zlib1g-dev libz-dev libseccomp-dev libcap-dev libminiupnpc-dev gettext
    sudo apt -y install libminiupnpc10 libzmq5
    sudo apt -y install libcanberra-gtk-module libqrencode-dev libzmq3-dev
    sudo apt -y install libqt5gui5 libqt5core5a libqt5webkit5-dev libqt5dbus5 qttools5-dev qttools5-dev-tools libprotobuf-dev protobuf-compiler
    sudo add-apt-repository -y ppa:bitcoin/bitcoin
    sudo apt -y update
    sudo apt install -y libdb4.8-dev libdb4.8++-dev libdb5.3 libdb5.3++


utils/screen-stratum.sh
#!/bin/bash
 STRATUM_DIR=/var/stratum
 
 screen -dmS groestl $STRATUM_DIR/run.sh groestl
 screen -dmS keccak $STRATUM_DIR/run.sh keccak
 screen -dmS neoscrypt $STRATUM_DIR/run.sh neo
 screen -dmS nist5 $STRATUM_DIR/run.sh nist5
 screen -dmS quark $STRATUM_DIR/run.sh quark
 screen -dmS scrypt $STRATUM_DIR/run.sh scrypt
 screen -dmS skein $STRATUM_DIR/run.sh skein
 screen -dmS x11 $STRATUM_DIR/run.sh x11
 screen -dmS x16rv2 $STRATUM_DIR/run.sh x16rv2
 screen -dmS c11 $STRATUM_DIR/run.sh c11
 screen -dmS deep $STRATUM_DIR/run.sh deep
 screen -dmS x11evo $STRATUM_DIR/run.sh x11evo
 screen -dmS x13 $STRATUM_DIR/run.sh x13
 screen -dmS x14 $STRATUM_DIR/run.sh x14
 screen -dmS x15 $STRATUM_DIR/run.sh x15
 screen -dmS x16r $STRATUM_DIR/run.sh x16r
 screen -dmS x17 $STRATUM_DIR/run.sh x17
 screen -dmS xevan $STRATUM_DIR/run.sh xevan
 screen -dmS timetravel $STRATUM_DIR/run.sh timetravel
 screen -dmS bitcore $STRATUM_DIR/run.sh bitcore
 screen -dmS hmq1725 $STRATUM_DIR/run.sh hmq1725
 screen -dmS tribus $STRATUM_DIR/run.sh tribus
 screen -dmS sha $STRATUM_DIR/run.sh sha
 screen -dmS sha256t $STRATUM_DIR/run.sh sha256t
 screen -dmS scryptn $STRATUM_DIR/run.sh scryptn
 screen -dmS luffa $STRATUM_DIR/run.sh luffa
 screen -dmS penta $STRATUM_DIR/run.sh penta
 screen -dmS qubit $STRATUM_DIR/run.sh qubit
 screen -dmS jha $STRATUM_DIR/run.sh jha
 screen -dmS dmd-gr $STRATUM_DIR/run.sh dmd-gr
 screen -dmS myr-gr $STRATUM_DIR/run.sh myr-gr
 screen -dmS lbry $STRATUM_DIR/run.sh lbry
 screen -dmS allium $STRATUM_DIR/run.sh allium
 screen -dmS lyra2 $STRATUM_DIR/run.sh lyra2
 screen -dmS lyra2v2 $STRATUM_DIR/run.sh lyra2v2
 screen -dmS zero $STRATUM_DIR/run.sh lyra2z
 screen -dmS blakecoin $STRATUM_DIR/run.sh blakecoin
 screen -dmS blake $STRATUM_DIR/run.sh blake
 screen -dmS blake2s $STRATUM_DIR/run.sh blake2s
 screen -dmS vanilla $STRATUM_DIR/run.sh vanilla
 screen -dmS decred $STRATUM_DIR/run.sh decred
 screen -dmS keccakc $STRATUM_DIR/run.sh keccakc
 screen -dmS phi $STRATUM_DIR/run.sh phi
 screen -dmS polytimos $STRATUM_DIR/run.sh polytimos
 screen -dmS whirlpool $STRATUM_DIR/run.sh whirlpool
 screen -dmS skein2 $STRATUM_DIR/run.sh skein2
 screen -dmS yescrypt $STRATUM_DIR/run.sh yescrypt
 screen -dmS yescryptR16 $STRATUM_DIR/run.sh yescryptR16
 screen -dmS zr5 $STRATUM_DIR/run.sh zr5
 screen -dmS sib $STRATUM_DIR/run.sh sib
 screen -dmS m7m $STRATUM_DIR/run.sh m7m
 screen -dmS veltor $STRATUM_DIR/run.sh veltor
 screen -dmS velvet $STRATUM_DIR/run.sh velvet
 screen -dmS argon2 $STRATUM_DIR/run.sh argon2

conf/prerequisite.sh
#!/bin/bash
#####################################################
# Source https://mailinabox.email/ https://github.com/mail-in-a-box/mailinabox
# Updated by cryptopool.builders for crypto use...
# Modified by Dirty Harry
#####################################################

echo
echo
echo -e "$CYAN => Check prerequisite : $COL_RESET"
echo

if [ "`lsb_release -d | sed 's/.*:\s*//' | sed 's/18\.04\.[0-9]/18.04/' `" == "Ubuntu 18.04 LTS" ]; then
  DISTRO=18
  sudo chmod g-w /etc /etc/default /usr
else [ "`lsb_release -d | sed 's/.*:\s*//' | sed 's/16\.04\.[0-9]/16.04/' `" != "Ubuntu 16.04 LTS" ];
  DISTRO=16
fi

#TOTAL_PHYSICAL_MEM=$(head -n 1 /proc/meminfo | awk '{print $2}')
#if [ $TOTAL_PHYSICAL_MEM -lt 1436000 ]; then
#  if [ ! -d /vagrant ]; then
#    TOTAL_PHYSICAL_MEM=$(expr \( \( $TOTAL_PHYSICAL_MEM \* 1024 \) / 1000 \) / 1000)
#    echo "Your Mining Pool Server needs more memory (RAM) to function properly."
#    echo "Please provision a machine with at least 1536 GB, 6 GB recommended."
#    echo "This machine has $TOTAL_PHYSICAL_MEM MB memory."
#    exit
#  fi
#fi

#if [ $TOTAL_PHYSICAL_MEM -lt 1436000 ]; then
#  echo "WARNING: Your Mining Pool Server has less than 1.5 GB of memory."
#  echo " It might run unreliably when under heavy load."
#fi

# Check swap
#echo Checking if swap space is needed and if so creating...

#SWAP_MOUNTED=$(cat /proc/swaps | tail -n+2)
#SWAP_IN_FSTAB=$(grep "swap" /etc/fstab)
#ROOT_IS_BTRFS=$(grep "\/ .*btrfs" /proc/mounts)
#TOTAL_PHYSICAL_MEM=$(head -n 1 /proc/meminfo | awk '{print $2}')
#AVAILABLE_DISK_SPACE=$(df / --output=avail | tail -n 1)
#if
#  [ -z "$SWAP_MOUNTED" ] &&
#  [ -z "$SWAP_IN_FSTAB" ] &&
#  [ ! -e /swapfile ] &&
#  [ -z "$ROOT_IS_BTRFS" ] &&
#  [ $TOTAL_PHYSICAL_MEM -lt 1536000 ] &&
#  [ $AVAILABLE_DISK_SPACE -gt 5242880 ]
#then
#  echo "Adding a swap file to the system..."

  # Allocate and activate the swap file. Allocate in 1KB chuncks
  # doing it in one go, could fail on low memory systems
#  sudo fallocate -l 3G /swapfile
#    if [ -e /swapfile ]; then
#      sudo chmod 600 /swapfile
#      hide_output sudo mkswap /swapfile
#      sudo swapon /swapfile
#      echo "vm.swappiness=10" >> sudo /etc/sysctl.conf
#    fi
# Check if swap is mounted then activate on boot
#  if swapon -s | grep -q "\/swapfile"; then
#    echo "/swapfile  none swap sw 0  0" >> sudo /etc/fstab
#  else
#    echo "ERROR: Swap allocation failed"
#  fi
#fi

ARCHITECTURE=$(uname -m)
if [ "$ARCHITECTURE" != "x86_64" ]; then
  if [ -z "$ARM" ]; then
    echo -e "$REDYiimP Install Script only supports x86_64 and will not work on any other architecture, like ARM or 32 bit OS. $COL_RESET"
    echo -e "$REDYour architecture is $ARCHITECTURE $COL_RESET"
    exit
  fi
fi

echo -e "$GREEN Done...$COL_RESET"